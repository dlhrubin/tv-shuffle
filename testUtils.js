import { fireEvent } from '@testing-library/react';
import axiosMock from 'axios';
import successSearch from './__fixtures__/successSearch';
import successShowInfo from './__fixtures__/successShowInfo';
import noPosterSearch from './__fixtures__/noPosterSearch';
import noPosterShowInfo from './__fixtures__/noPosterShowInfo';
import notFoundSearch from './__fixtures__/notFoundSearch';
import noDataSearch from './__fixtures__/noDataSearch';
import noDataShowInfo from './__fixtures__/noDataShowInfo';

jest.mock('axios');

// Mock successful API call (returns show)
export const successfulCall = (getByTestId) => {
  fireEvent.change(getByTestId('search-bar'), { target: { value: 'how i met' } });
  axiosMock.get.mockResolvedValueOnce(successSearch);
  axiosMock.get.mockResolvedValueOnce(successShowInfo);
  fireEvent.submit(getByTestId('search-form'));
};

// Mock successful API call (returns show but no poster)
export const noPosterCall = (getByTestId) => {
  fireEvent.change(getByTestId('search-bar'), { target: { value: 'days of de' } });
  axiosMock.get.mockResolvedValueOnce(noPosterSearch);
  axiosMock.get.mockResolvedValueOnce(noPosterShowInfo);
  fireEvent.submit(getByTestId('search-form'));
};

// Mock unsuccessful API call (show title not found)
export const notFoundCall = (getByTestId) => {
  fireEvent.change(getByTestId('search-bar'), { target: { value: 'abcd' } });
  axiosMock.get.mockResolvedValueOnce(notFoundSearch);
  fireEvent.submit(getByTestId('search-form'));
};

// Mock unsuccessful API call (insufficient data found for show)
export const noDataCall = (getByTestId) => {
  fireEvent.change(getByTestId('search-bar'), { target: { value: 'days of' } });
  axiosMock.get.mockResolvedValueOnce(noDataSearch);
  axiosMock.get.mockResolvedValueOnce(noDataShowInfo);
  fireEvent.submit(getByTestId('search-form'));
};
