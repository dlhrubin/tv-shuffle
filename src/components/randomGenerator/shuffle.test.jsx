import React from 'react';
import { render, cleanup, waitFor } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import { successfulCall } from '../../../testUtils';
import Shuffle from './shuffle';
import RandomGenerator from '../randomGenerator';

afterEach(cleanup);

describe('Shuffle', () => {
  it('renders without crashing', () => {
    const { asFragment } = render(<Shuffle />);
    expect(asFragment(<Shuffle />)).toMatchSnapshot();
  });
  it('is enabled when search bar returns result', async () => {
    const { getByTestId } = render(<RandomGenerator />);
    successfulCall(getByTestId);
    await waitFor(() => {
      expect(getByTestId('shuffle-button')).toBeEnabled();
    });
  });
});
