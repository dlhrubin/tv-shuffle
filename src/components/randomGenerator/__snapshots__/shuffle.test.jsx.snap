// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Shuffle renders without crashing 1`] = `
<DocumentFragment>
  <section
    class="shuffle"
    id="shuffle"
  >
    <span>
      Shuffle
    </span>
    <button
      aria-label="Shuffle"
      data-testid="shuffle-button"
      disabled=""
      type="button"
    >
      <i
        class="fas fa-dice"
      />
    </button>
  </section>
</DocumentFragment>
`;
